File: api/src/StudyHive.Api/Data/Entities/S1/StudentProfile.cs
namespace StudyHive.Api.Data.Entities;

public class StudentProfile
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public required string StudentNumber { get; set; }
    public required string Department { get; set; }
    public int YearOfStudy { get; set; }
    public int MaxBookingsPerWeek { get; set; } = 3;
    public int PenaltyPoints { get; set; }
    public DateOnly? SuspendedUntil { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public User User { get; set; } = null!;
    public ICollection<BookingRequest> BookingRequests { get; set; } = new List<BookingRequest>();
}

File: mobile/lib/models/student_profile.dart
/// Mirrors StudyHive.Api's StudentProfileResponse.
class StudentProfile {
  final String id;
  final String userId;
  final String studentNumber;
  final String department;
  final int yearOfStudy;
  final int maxBookingsPerWeek;
  final int penaltyPoints;
  final String? suspendedUntil;
  final bool isActive;

  const StudentProfile({
    required this.id,
    required this.userId,
    required this.studentNumber,
    required this.department,
    required this.yearOfStudy,
    required this.maxBookingsPerWeek,
    required this.penaltyPoints,
    required this.suspendedUntil,
    required this.isActive,
  });

  factory StudentProfile.fromJson(Map<String, dynamic> json) => StudentProfile(
        id: json['id'] as String,
        userId: json['userId'] as String,
        studentNumber: json['studentNumber'] as String,
        department: json['department'] as String,
        yearOfStudy: json['yearOfStudy'] as int,
        maxBookingsPerWeek: json['maxBookingsPerWeek'] as int,
        penaltyPoints: json['penaltyPoints'] as int,
        suspendedUntil: json['suspendedUntil'] as String?,
        isActive: json['isActive'] as bool,
      );
}
